### PR TITLE
More support for 3D biomes

### PR DESCRIPTION
**Pull request by MattBDev**
Adds more support for 3D Biomes. This was all done manually since cherry-picking failed.

### CODE REVIEW

File: worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/EmptyClipboard.kt [hunk 1]
```diff
@@ -64,7 +64,11 @@ object EmptyClipboard : Clipboard {
         return false
     }
 
-    override fun setBiome(position: BlockVector2, biome: BiomeType): Boolean {
+    override fun setBiome(position: BlockVector3, biome: BiomeType): Boolean {
```
Comment (N0tMyFaultOG) [inline worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/EmptyClipboard.kt:67]: I'm still surprised why we have random Kotlin classes here and there.

**Review by N0tMyFaultOG — APPROVED** [review]

File: worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/EmptyClipboard.kt (hunk 1 above)
Comment (MattBDev) [reply worldedit-core/src/main/java/com/boydti/fawe/object/clipboard/EmptyClipboard.kt:67]: I'm still interested in exploring potential uses for unsigned integers.

**Review by aurorasmiles — APPROVED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/extent/buffer/ForgetfulExtentBuffer.java [hunk 2]
```diff
@@ -116,22 +111,22 @@ public ForgetfulExtentBuffer(Extent delegate, Mask mask) {
...
-            max2d = position;
+        if (max == null) {
+            max = position;
         } else {
-            max2d = max2d.getMaximum(position);
+            max = max.getMaximum(position);
         }
 
-        if (biomeMask.test(getExtent(), position)) {
+        if (mask.test(getExtent(), position)) {
```
Comment (aurorasmiles) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extent/buffer/ForgetfulExtentBuffer.java:129]: Please fix the merge conflict here

File: worldedit-core/src/main/java/com/sk89q/worldedit/extent/buffer/ForgetfulExtentBuffer.java [hunk 3]
```diff
@@ -142,21 +137,21 @@ public boolean setBiome(BlockVector2 position, BiomeType biome) {
...
+        if (max == null) {
+            max = BlockVector3.at(x, y, z);
         } else {
-            max2d = max2d.getMaximum(BlockVector2.at(x,z));
+            max = max.getMaximum(BlockVector3.at(x, y, z));
         }
 
-        if (biomeMask.test(getExtent(), BlockVector2.at(x,z))) {
-            biomeBuffer.put(BlockVector2.at(x,z), biome);
+        if (mask.test(getExtent(), BlockVector3.at(x, y, z))) {
```
Comment (aurorasmiles) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extent/buffer/ForgetfulExtentBuffer.java:153]: I believe this also conflicts

**Review by aurorasmiles — CHANGES_REQUESTED** [review]

**Review by aurorasmiles — APPROVED** [review]

**Pull request merged**